### PR TITLE
Support for new installer file name format

### PR DESCRIPTION
**Pull request by alameenshah**
- Support for New Asset installer name format.
- New GVFSPlatform constant for installer file name extension.
- Renamed installer file name from "SetupGVFS.version.exe" to "VFSGit.version.exe"
- Updated UT.

New file name format -
``` 
Git Mac – Git-2.17.1.gvfs.2.5.g2962052-64-bit.dmg
Git Win - Git-2.17.1.gvfs.2.5.g2962052-64-bit.exe (No name change for Git).
GVFS Mac - VFSGit.1.0.18234.1.dmg
GVFS Windows - VFSGit.1.0.18234.1.exe
```

#### Testing Note - 
This PR will cause GVFS upgrade to fail. This is because the GVFS installer file name is in the old format (SetupGVFS.1.0.18234.1.exe) in the last released GVFS release.

### CODE REVIEW

File: GVFS/GVFS.Platform.Mac/MacPlatform.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace GVFS.Platform.Mac
     public partial class MacPlatform : GVFSPlatform
     {
         public MacPlatform()
-            : base(executableExtension: string.Empty)
+            : base(executableExtension: string.Empty, installerExtension: string.Empty)
```
Comment (wilbaker) [inline GVFS/GVFS.Platform.Mac/MacPlatform.cs]: Should `installerExtension` be .dmg for Mac?

File: GVFS/GVFS.UnitTests/Mock/Common/MockPlatform.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace GVFS.UnitTests.Mock.Common
     public class MockPlatform : GVFSPlatform
     {
         public MockPlatform()
-            : base(executableExtension: ".mockexe")
+            : base(executableExtension: ".mockexe", installerExtension: ".exe")
```
Comment (wilbaker) [inline GVFS/GVFS.UnitTests/Mock/Common/MockPlatform.cs]: For consistency with `executableExtension` in the mock, update `installerExtension` to `".mockexe"` as well?

File: GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs [hunk 3]
```diff
@@ -90,7 +90,6 @@ public void Execute()
                     {
                         mountError = Environment.NewLine + "WARNING: " + mountError;
                         this.output.WriteLine(mountError);
-                        this.ExitCode = ReturnCode.Success;
```
Comment (wilbaker) [inline GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs]: Does this change also need to go into the other auto-upgrade feature branch?

File: GVFS/GVFS.Common/ProductUpgrader.Shared.cs [hunk 4]
```diff
@@ -11,7 +11,7 @@ public partial class ProductUpgrader
         public const string DownloadDirectory = "Downloads";
 
         private const string RootDirectory = UpgradeDirectoryName;
-        private const string GVFSInstallerFileNamePrefix = "SetupGVFS";
+        private const string GVFSInstallerFileNamePrefix = "VFSGit";
```
Comment (sanoursa) [inline GVFS/GVFS.Common/ProductUpgrader.Shared.cs]: Are you planning on updating the actual installer in this PR too? If not, do we need to support both format?

File: GVFS/GVFS.Platform.Mac/MacPlatform.cs (hunk 1 above)
Comment (alameenshah) [reply GVFS/GVFS.Platform.Mac/MacPlatform.cs]: Right. Updated.

File: GVFS/GVFS.UnitTests/Mock/Common/MockPlatform.cs (hunk 2 above)
Comment (alameenshah) [reply GVFS/GVFS.UnitTests/Mock/Common/MockPlatform.cs]: Updated.

File: GVFS/GVFS.UnitTests/Common/GitVersionTests.cs [hunk 5]
```diff
@@ -10,9 +11,9 @@ public class GitVersionTests
         [TestCase]
         public void TryParseInstallerName()
         {
-            this.ParseAndValidateInstallerVersion("Git-1.2.3.gvfs.4.5.gb16030b-64-bit.exe");
-            this.ParseAndValidateInstallerVersion("git-1.2.3.gvfs.4.5.gb16030b-64-bit.exe");
-            this.ParseAndValidateInstallerVersion("Git-1.2.3.gvfs.4.5.gb16030b-64-bit.EXE");
+            this.ParseAndValidateInstallerVersion("Git-1.2.3.gvfs.4.5.gb16030b-64-bit" + GVFSPlatform.Instance.Constants.InstallerExtension);
+            this.ParseAndValidateInstallerVersion("git-1.2.3.gvfs.4.5.gb16030b-64-bit" + GVFSPlatform.Instance.Constants.InstallerExtension);
+            this.ParseAndValidateInstallerVersion("Git-1.2.3.gvfs.4.5.gb16030b-64-bit" + GVFSPlatform.Instance.Constants.InstallerExtension);
```
Comment (nickgra) [inline GVFS/GVFS.UnitTests/Common/GitVersionTests.cs]: Does the third line in this test actually test anything now that we're using a constant?

Comment (alameenshah) [reply GVFS/GVFS.UnitTests/Common/GitVersionTests.cs]: Yes - it tests the TryParseInstallerName() in GitVersion.cs. That method has been updated to accept a Platform dependent InstallerExtension.

File: GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs (hunk 3 above)
Comment (alameenshah) [reply GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs]: Yes... It is not related to name change I know, I just happened to discover it while testing this PR.

File: GVFS/GVFS.Common/ProductUpgrader.Shared.cs (hunk 4 above)
Comment (alameenshah) [reply GVFS/GVFS.Common/ProductUpgrader.Shared.cs]: >>Are you planning on updating the actual installer in this PR too?

Yes - renamed the actual installer.

**Pull request merged**